src/com/facebook/buck/apple/project_generator/SchemeGenerator.java

```diff
@@ -336,18 +369,32 @@ public static Element serializeLaunchAction(Document doc, XCScheme.LaunchAction
     launchActionElem.setAttribute(
         "launchStyle", launchStyle == XCScheme.LaunchAction.LaunchStyle.AUTO ? "0" : "1");
 
+    if (launchAction.getEnvironmentVariables().isPresent()) {
+      Element environmentVariablesElement =
+          serializeEnvironmentVariables(doc, launchAction.getEnvironmentVariables().get());
+      launchActionElem.appendChild(environmentVariablesElement);
+    }
+
     return launchActionElem;
   }
 
   public static Element serializeProfileAction(Document doc, XCScheme.ProfileAction profileAction) {
     Element profileActionElem = doc.createElement("ProfileAction");
+    profileActionElem.setAttribute("shouldUseLaunchSchemeArgsEnv", "YES");
```

Comment:
Why do you have to set it explicitly here?

Reply:
This is already present in the Test action. This is to mimic Xcode's default settings of having the same variables on test/profile as launch unless you explicitly specify otherwise.